### PR TITLE
Refactor load save

### PR DESCRIPTION
**Pull request by denys-hrulov**
Remove TaskDataConverter.
Remove interfaces of ModelHolder and ModelCreator.
Remove StreamOwner and inject stream in persisiter on construcor.
Create PersisterCreator class.
Rename TaskModel to TaskRepository.
Move creation of default TaskService from API to core root.

### CODE REVIEW

**Review by IlliaDmitriiev — APPROVED** [review]
You should try to make PR a little bit smaller.

**Comment by denys-hrulov** [comment]
> You should try to make PR a little bit smaller.

It's hard

File: src/core/RepositoryHolder.cpp [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    auto file = std::make_shared<std::fstream>(filepath, std::ios::in);
+    if (!file->is_open()) {
+        return false;
+    }
+    auto new_model = creator_->CreateModel();
+    auto persister = persister_creator_->CreatePersister(*new_model, file);
+    if (!persister->Load()) {
+        return false;
+    }
+    std::swap(model_, new_model);
```
Comment (denys-hrulov) [inline src/core/RepositoryHolder.cpp:31]: wtf

**Comment by IliaIlyin** [comment]
https://rozetka.com.ua/143187578/p143187578/?gclid=Cj0KCQjwqKuKBhCxARIsACf4XuFWM4Fy_txJp9mGdvoyLpyhG0qZLBOAHn9RbV9kQ4dND5oYH0pzWC0aAuqJEALw_wcB

**Pull request merged**